src/lib/index/server.js

```diff
@@ -25,236 +25,211 @@
+			Object.defineProperty(me, PROPERTY_SERVER, { value: server });
+
+			// Add the publisher
+			Object.defineProperty(me, PROPERTY_PUBLISHER, { value: new Publisher(config) });
+
+			// Define the router configuration for the server
+			Object.defineProperty(me, PROPERTY_ROUTER_CONFIGURATION, { value: {} });
+
+			// Define the route configuration for the server
+			Object.defineProperty(me, PROPERTY_ROUTE_CONFIGURATION, { value: {} });
+
+			// Define the route validator
+			Object.defineProperty(me, PROPERTY_ROUTE_VALIDATOR, { value: new RouteValidator() });
+
+		} catch (e) {
+			me.error(e);
+			throw e;
+		}
+
+	}
 
 	/**
-	 * Adds a 'route' to the server
+	 * Adds a 'route' to the server.
 	 * 
-	 * @example
-	 * // adds schemas to the default ( http://host/{schemaName} ) route
-	 * server.addRoute({
-	 * 	schemaNameToDefinition: {
-	 * 		test: {} // an avro compliant schema (see avro schema API)
-	 * 	}
-	 * });
-	 * @example
-	 * // adds schemas to a route at ( http://host/api/test/{schemaName} )
-	 * server.addRoute({ schemaNameToDefinition, apiEndpoint: '/api/test' });
-	 * @example
-	 * // adds middleware functions contained in the middlewares array (see express middleware API)
-	 * server.addRoute({ schemaNameToDefinition, middlewares: [...] });
-	 * 
-	 * @param {object} config - { schemaNameToDefinition [, middlewares] [, apiEndpoint] [, responseWriter] }
-	 * @param {object} config.schemaNameToDefinition - schema name to definition map
-	 * @param {object} config.middlewares - middleware functions (optional)
-	 * @param {object} config.apiEndpoint - api endpoint (optional, default: /)
-	 * @param {ResponseWriter} config.responseWriter - a function to handle the writing of responses.
+	 * @param {Object} route - The route.
+	 * @param {string|Object} route.schema - The Apache Avro schema for this route.
+	 * @param {string} route.method=POST - The method to use for this route.
+	 * @param {Function[]} route.middlewares - The middleware functions for this route.
+	 * @param {Function} route.responseWriter - The function to use before writing the response.
+	 * @param {string} [route.endpoint=/] - The API endpoint.
 	 * 
-	 * @returns {Server}
+	 * @returns {Server} The server.
 	 */
-	addRoute({ schemaNameToDefinition, middlewares, apiEndpoint, responseWriter }) {
+	addRoute(config) {
 
-		// create router
+		// Validate the route configuration.
+		this[PROPERTY_ROUTE_VALIDATOR].validate(config);
+
+		// Now we know the configuration is valid, add the route.
 		const
-			publisher = publisherStore[this],
+			me = this,
+			schema = config.schema,
+			responseWriter = config.responseWriter,
+			fullSchemaName = schema.name,
+			schemaNameParts = fullSchemaName.split('.'),
+			schemaNamespace = _.initial(schemaNameParts).join('.'),
+			schemaName = _.last(schemaNameParts),
+			apiEndpoint = `${config.endpoint}${schemaNamespace}`.replace(/\./g, '/'),
+
+			routeConfiguration = _.get(me[PROPERTY_ROUTE_CONFIGURATION], apiEndpoint, {});
```

Comment:
Works fine as is, but you're effectively splitting the path out over the first and second args.
I'd probably do
```javascript
_.get(me, `${PROPERTY_ROUTE_CONFIGURATION}.apiEndpoint`, {})
```

Reply:
Done